app/src/main/java/com/zeapo/pwdstore/git/BaseGitActivity.kt

```diff
@@ -0,0 +1,183 @@
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import com.zeapo.pwdstore.git.config.ConnectionMode
+import com.zeapo.pwdstore.git.config.Protocol
+import com.zeapo.pwdstore.git.config.SshApiSessionFactory
+import com.zeapo.pwdstore.utils.PasswordRepository
+import java.io.File
+import timber.log.Timber
+
+/**
+ * Abstract AppCompatActivity that holds some information that is commonly shared across git-related
+ * tasks and makes sense to be held here.
+ */
+abstract class BaseGitActivity : AppCompatActivity() {
+    lateinit var protocol: Protocol
+    lateinit var connectionMode: ConnectionMode
+    lateinit var hostname: String
+    lateinit var serverUrl: String
+    lateinit var serverPort: String
+    lateinit var serverUser: String
+    lateinit var serverPath: String
+    lateinit var username: String
+    lateinit var email: String
+    var identityBuilder: SshApiSessionFactory.IdentityBuilder? = null
+    var identity: SshApiSessionFactory.ApiIdentity? = null
+    lateinit var settings: SharedPreferences
+        private set
+
+    @CallSuper
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        settings = PreferenceManager.getDefaultSharedPreferences(this)
+        protocol = Protocol.fromString(settings.getString("git_remote_protocol", null))
+        connectionMode = ConnectionMode.fromString(settings.getString("git_remote_auth", null))
+        serverUrl = settings.getString("git_remote_server", null) ?: ""
+        serverPort = settings.getString("git_remote_port", null) ?: ""
+        serverUser = settings.getString("git_remote_username", null) ?: ""
+        serverPath = settings.getString("git_remote_location", null) ?: ""
+        username = settings.getString("git_config_user_name", null) ?: ""
+        email = settings.getString("git_config_user_email", null) ?: ""
+        updateHostname()
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            android.R.id.home -> {
+                finish()
+                true
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
+    override fun onDestroy() {
+        // Do not leak the service connection
+        if (identityBuilder != null) {
+            identityBuilder!!.close()
+            identityBuilder = null
+        }
+        super.onDestroy()
+    }
+
+    /**
+     * Update the [hostname] field with the values that build it up. This function returns a boolean
+     * indicating whether or not the values are valid or not, and only adds the `origin` remote when
+     * it is.
+     */
+    fun updateHostname(): Boolean {
+        var valid = false
+        hostname = when (protocol) {
+            Protocol.Ssh -> {
+                val hostname = StringBuilder()
+                hostname.append("$serverUser@${serverUrl.trim { it <= ' '}}:")
+                if (serverPort == "22") {
```

Comment:
I am unable to save my config if I enter 22 as the port, I haven't understood why though.